cirq/contrib/acquaintance/gates_test.py

```diff
@@ -124,10 +132,35 @@ def test_swap_network_gate():
     """.strip()
     assert actual_text_diagram == expected_text_diagram
 
+@pytest.mark.parametrize('part_lens',
+    [tuple(randint(1, 3) for _ in range(randint(2, 10))) for _ in range(3)])
+def test_acquaint_part_pairs(part_lens):
+    n_qubits = sum(part_lens)
+    qubits = tuple(cirq.NamedQubit(s) for s in alphabet)[:n_qubits]
+    swap_network_op = SwapNetworkGate(
+        part_lens, acquaintance_size=None)(*qubits)
+    swap_network = cirq.Circuit.from_ops(
+            swap_network_op, device=UnconstrainedAcquaintanceDevice)
+    initial_mapping = {q: i for i, q in enumerate(qubits)}
+
+    actual_opps = get_logical_acquaintance_opportunities(
+            swap_network, initial_mapping)
+    parts = []
+    q = 0
```

Comment:
Move this loop to the top, and merge `q` into `n_qubits`

Reply:
Done